.github/workflows/build_and_test_dbt.yaml

```diff
@@ -33,21 +33,35 @@ jobs:
       - name: Configure dbt environment
         uses: ./.github/actions/configure_dbt_environment
 
-      - name: Cache dbt state directory
+      # We have to use the separate `restore`/`save` actions instead of the
+      # unified `cache` action because only `restore` provides access to the
+      # `cache-matched-key` and `cache-primary-key` outputs as of v3
```

Comment:
This is a frustrating difference! Note [the outputs of the `restore` action](https://github.com/actions/cache/blob/f7ebb81a3f195b4fb88dab7c14e2f7aff52045aa/restore/action.yml#L26-L32) vs. [the outputs of the `cache` action](https://github.com/actions/cache/blob/f7ebb81a3f195b4fb88dab7c14e2f7aff52045aa/action.yml#L12).